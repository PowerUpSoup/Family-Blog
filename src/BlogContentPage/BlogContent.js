import React, { Component } from 'react';
import Comment from '../Comment/Comment';
import ApiContext from '../ApiContext';
import CommentForm from '../CommentForm/CommentForm';
import { Link } from 'react-router-dom';

class BlogContent extends Component {
    static contextType = ApiContext;
    
    render() {
        const { articles = [] } = this.context
        const articleid = parseInt(this.props.match.params.id)
        const article = articles.find(article => article.id === articleid) || { content: '', title: '' }
        const isLoggedIn = this.context.isLoggedIn
        return (
            <div className="blog-Content-Page">
                <div className="blog-post">
                    <header>
                        <h1>{article.title}</h1>
                    </header>
                    <div>
                        <pre className="blog-post-content">{article.content}</pre>
                    </div>
                </div>
                
                {isLoggedIn
                    ? <CommentForm articleid={articleid}/>
                    : <Link to='/login'>Login to leave a comment!</Link>}

                <section className="comment-section">
                    <div className="comments">
                        {this.context.comments.map((comment, key) => {
                            if (comment.articleid === article.id) {
                                return <Comment
                                    key={key}
                                    comment={comment} />
                            } else {
                                return null
                            }
                        })}
                    </div>
                </section>
            </div>
        )
    }
}

export default BlogContent;