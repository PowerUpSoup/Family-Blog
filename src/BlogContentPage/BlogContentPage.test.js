import React from 'react';
import BlogContent from './BlogContent';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'


const props = {
        match: {params: {id: 1}, isExact: true, path: "", url: ""},
    }

test('renders learn react link', () => {
      const wrapper = shallow(
        <BrowserRouter>
            <BlogContent {...props} />
        </BrowserRouter>
      )
      expect(toJson(wrapper)).toMatchSnapshot()
});