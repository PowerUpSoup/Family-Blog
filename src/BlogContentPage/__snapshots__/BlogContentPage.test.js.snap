// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <BlogContent
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": 1,
        },
        "path": "",
        "url": "",
      }
    }
  />
</Router>
`;
